profiles/tests/test_models.py

```diff
@@ -0,0 +1,163 @@
+from rest_framework.test import APITestCase, APIClient
+from authentication.models import CustomUser
+from profiles.models import Activity, Category, Profile
+from rest_framework import status
+
+
+class SavedCompaniesListCreateDestroyAPITest(APITestCase):
+
+    @classmethod
+    def setUpTestData(cls):
+        test_user = CustomUser.objects.create_user(
+            person_email="testuser1@test.com",
+            password="testuser1",
+            person_name="testuser1",
+            person_surname="testuser1",
+            comp_name="testuser1",
+            comp_registered=True,
+            comp_is_startup=False
+        )
+
+        test_user2 = CustomUser.objects.create_user(
+            person_email="testuser2@test.com",
+            password="testuser2",
+            person_name="testuser2",
+            person_surname="testuser2",
+            comp_name="testuser2",
+            comp_registered=True,
+            comp_is_startup=False
+        )
+
+        test_activity = Activity.objects.create(activity_id=1, name="Test Activity")
+        test_category = Category.objects.create(category_id=1, name="Test Category")
+        test_company1 = Profile.objects.create(
+            comp_official_name="Test Company",
+            comp_region="Test Region",
+            comp_common_info="Test Common Info",
+            comp_phone_number="1234567890",
+            comp_EDRPOU=1234567890,
+            comp_year_of_foundation=2020,
+            comp_service_info="Test Service Info",
+            comp_product_info="Test Product Info",
+            comp_address="Test Address",
+            comp_banner_image=None,
+            person=test_user,
+            person_position="Test Position",
+            startup_idea="Test Startup Idea"
+            )
+        test_company2 = Profile.objects.create(
+            comp_official_name="Test Company2",
+            comp_region="Test Region2",
+            comp_common_info="Test Common Info2",
+            comp_phone_number="1234527890",
+            comp_EDRPOU=1234527890,
+            comp_year_of_foundation=2022,
+            comp_service_info="Test Service Info2",
+            comp_product_info="Test Product Info2",
+            comp_address="Test Address2",
+            comp_banner_image=None,
+            person=test_user2,
+            person_position="Test Position2",
+            startup_idea="Test Startup Idea2"
+            )
+
+    def setUp(self):
+        self.client = APIClient()
+        self.url = '/api/saved-list/'
+
+        # login users & get tokens
+        self.token1 = self.client.post(
+            path="/api/auth/token/login/",
+            data={
+                "person_email": "testuser1@test.com",
+                "password": "testuser1"
+            }).data["auth_token"]
+        self.client.credentials(HTTP_AUTHORIZATION=f"Token {self.token1}")
+        self.user1_id = self.client.get(
+            path="http://127.0.0.1:8000/api/auth/users/me/").data["id"]
```

Comment:
1. Use relative URLs `/api/...`.
2. Tokens clearly look as pain in the ass. Again, I recommend to not use them. With ordinary authentication it's as easy as just `self.client.force_login(user_instance)`.
3. I strongly recommend to authenticate users in-place where it's needed. I.e. in each testcase, instead of `setUp` etc.

Reply:
At the moment, I consider it unacceptable to implement logging + tokens in each test.
If the tokens are removed later, I will rewrite the tests